src/algebra/char_p/basic.lean

```diff
@@ -35,6 +35,10 @@ theorem char_p.cast_eq_zero [add_monoid R] [has_one R] (p : ℕ) [char_p R p] :
   (p:R) = 0 :=
 (char_p.cast_eq_zero_iff R p p).2 (dvd_refl p)
 
+lemma char_ne_zero_of_coe_zero {R : Type u} [add_monoid R] [has_one R] (p : ℕ) [char_p R p]
```

Comment:
This looks a little specific to me (and the name is almost as long as the proof). Do you have any application in mind?

Reply:
I'm currently  creating a PR with definition & theorems on equal/mixed characteristics and I use it there with `char_p.char_is_prime_or_zero` (with says `nat.prime p ∨ p = 0`) and `or_iff_not_imp_right`. But as you pointed out, as the proof isn't that long, I could just copy-paste it in there instead of having a separate lemma for it.